Formula/s6.rb

```diff
@@ -0,0 +1,39 @@
+class S6 < Formula
+  desc "Small & secure supervision software suite."
+  homepage "http://skarnet.org/software/s6/"
+  url "http://skarnet.org/software/s6/s6-2.3.0.0.tar.gz"
+  sha256 "49c9a0476d93c6279c776ca5ae9c49b3fd39c356eda674828aebf78f8488e8e6"
+  head "git://git.skarnet.org/s6"
+
+  depends_on "skalibs"
+  depends_on "execline"
+
+  def install
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--bindir=#{libexec}/bin",
+      "--sbindir=#{libexec}/sbin",
+      "--with-sysdeps=#{Formula["skalibs"].opt_prefix}/lib/skalibs/sysdeps",
+      "--with-lib=#{Formula["skalibs"].opt_prefix}/lib",
+      "--with-lib=#{Formula["execline"].opt_prefix}/lib/execline",
+      "--disable-shared"
+    system "make", "install"
+
+    # Some S6 tools expect execline binaries to be on the path
```

Comment:
Unlike other shells, execline doesn't have any real builtins; instead, it comes with external commands. 

- Many of these are original (`multidefine`, `forbacktickx`), no problem there
- Others (such as `cd`, `exec`, `umask`) overlap with shell builtins, but have different semantics  than what you would expect (since execline relies on Bernstein chaining); this is actually also fine, since they are caught by shell builtins, so won't be picked from the path.
- One command is problematic: `import`; this one conflicts with ImageMagick.

So I had the choice of putting everything except `import` in PATH, or only `execline`, and make `execline` set the path before executing its script. Since these commands are probably mostly only used in the context of `execline` scripts, I took the second option. However, `s6` is an exception, as it also uses some execline binaries here and there, so I had to also set the path there.

I'm open to taking the other path, but it would mean `import` is either not available to execline scripts (which is a bit annoying).